xds/internal/balancer/cdsbalancer/cdsbalancer.go

```diff
@@ -430,6 +430,20 @@ func (b *cdsBalancer) ExitIdle() {
 	})
 }
 
+// Node ID needs to be manually added to errors generated in the following
+// scenarios:
+//   - resource-does-not-exist: since the xDS watch API uses a separate callback
+//     instead of returning an error value,
+//   - received a good update from the xDS client, but the update either contains
+//     an invalid security configuration or contains invalid aggragate cluster
+//     config.
```

Comment:
Just to sanity check - these are things not possible to validate at resource parsing time, right?

Reply:
Yes.